docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+
+   .. code:: java
+
+      public class Item {
+          private String name;
+          private int weight;
+          private int value;
+
+          public Item(String name, int weight, int value) {
+              this.name = name;
+              this.weight = weight;
+              this.value = value;
+          }
+
+          public String getName() {
+              return name;
+          }
+
+          public int getWeight() {
+              return weight;
+          }
+
+          public int getValue() {
+              return value;
+          }
+      }
+
+
+-  **Static questioning:** Avoid giving students exercises that test
+   knowledge in a static manner. Make the newly learned concept part of a
+   bigger picture and incorporate it with previously learned concepts
+   (goes hand in hand with working on a project throughout the
+   semester). Exercises that were previously solved statically.
+
+   -  Stream functionality
+
+   -  Recursion
+
+   .. figure:: programming/AI-proof-exercises/Streams-exercise.png
+      :align: center
+      :width: 50%
+
+      Part of a static stream exercise where a method's functionality needed to
+      be created using only streams.
+
+   An alternate formulation for this exercise which promotes critical thinking and prevents
+   blind copy and paste attempts could look like this:
+
+      In the Artemis class, you're faced with a collection of exams represented by a stream,
+      each having a status and a mode of taking - either online (``isOnline=true``) or onsite (``isOnline=false``).
+
+      Your challenge is to dive deep into this collection and uncover some interesting statistics. Here's a
+      hint of what you should be looking for:
+
+      - Are all exams in a worthy state to be considered for your analysis? Think about the "VALID" status.
+
+      - We're living in a digital age, but not all exams have embraced the online platform. Can you gauge the distribution?
+
+      - When presenting your findings, a map could be an elegant way to represent the dichotomy of online and onsite exams.
+
+      Remember, it's not just about counting. It's about discerning patterns and presenting them meaningfully. If, hypothetically, you discovered that all exams were conducted online, how would your result reflect that?
+
+   .. note::
+      It might be prudent not to specifically mention in the problem statement that streams should be used for the exercise.
+      Rather state that the students should use the concepts that were
+      introduced in the last week and rename the exercise to something like "Exam System with Streams."
+      This recommendation can be applied at any level to every exercise type.
+
+-  **Additional resources:** Give the students the opportunity to read
+   more about concepts and problems they struggle with. Provide
+   additional links and resources for specific problems. Some examples
+   for resources that can be provided for suitable exercises:
```

Comment:
This is always a good idea, but it does not really help to protect against LLMs. LLMs probably already know about these algorithms, unless they are very niche. If this was not covered in the lecture it would only make the exercise harder for the students.